### DISCUSSION TITLE
A new name?

### DISCUSSION THREAD

**Discussion by aaronmw** [graz-sh/graz#9] [Q&A] [answered]
Just curious: a) where the name comes from and b) whether there's appetite for changing it, or considering alternatives.

These hooks could really lower the barrier to entry for building front-end stuff in Cosmos, but the name isn't doing it any favours re: discoverability. Even prefixing `react-` onto the library name per React conventions could be helpful, maybe?? Nit-picky, but wanted to have the discussion!

**Comment by grikomsn** [comment, marked as answer]
Thank you for the interest, @aaronmw! 😄 

Previously me and @codingki named this project `wadesta` "because reasons"[^1], then after internal meetings and brainstorming with the engineering team we decide the new name will be `graz`, which is a name of the city where Johannes Kepler (yes, as in the [Keplr Wallet](https://keplr.app)) teach mathematics at a seminary school[^2].

Re: discoverability, some teammates do agree that having a generic name would be better. Here is one of the feedbacks from our internal discussion:

> I tend to default to the arguably boring, but also explicit. `react-cosmos-hooks` or something ...
>
> On the other side, we have libs like [`wagmi`](https://wagmi.sh/) which could have been called `react-ethereum-hooks`. This kind of naming is more fun to brand, etc. but also means people wont know what it does at first glance.

As the project author, I prefer having a short and unique name so it sticks like `wagmi`. But if the community prefers something around `react-cosmos-hooks`, `use-keplr`, or anything, we would consider it in the future. In the meantime, we're focusing on adding functionalities and working with various cosmos developers to improve the library.

Again, thank you for the interest!

[^1]: https://www.youtube.com/watch?v=02ryL1zNsho
[^2]: https://en.wikipedia.org/wiki/Johannes_Kepler

**Reply by aaronmw** [reply]
I hadn't heard of `wagmi` but seeing it as `react-ethereum-hooks` got me instantly psyched about it 😜 Anyway, I appreciate that naming is low on your list. When you do go seeking adoption though, I think that name change would be a huge boon. `useKeplr` is 👌
